apps/server/src/shared/infra/mail/mail.service.ts

```diff
@@ -10,12 +11,53 @@ interface MailServiceOptions {
 
 @Injectable()
 export class MailService {
+	private readonly domainBlacklist: string[];
+
 	constructor(
 		private readonly amqpConnection: AmqpConnection,
-		@Inject('MAIL_SERVICE_OPTIONS') private readonly options: MailServiceOptions
-	) {}
+		@Inject('MAIL_SERVICE_OPTIONS') private readonly options: MailServiceOptions,
+		private readonly configService: ConfigService<IMailConfig, true>
+	) {
+		this.domainBlacklist = this.configService.get<string[]>('ADDITIONAL_BLACKLISTED_EMAIL_DOMAINS');
+	}
 
 	public async send(data: Mail): Promise<void> {
+		if (this.domainBlacklist.length > 0) {
+			data.recipients = this.removeEmailAddressesThatHaveBlacklistedDomain(
+				data.recipients,
+				this.domainBlacklist
+			) as string[];
+			data.cc = this.removeEmailAddressesThatHaveBlacklistedDomain(data.cc, this.domainBlacklist);
+			data.bcc = this.removeEmailAddressesThatHaveBlacklistedDomain(data.bcc, this.domainBlacklist);
+			data.replyTo = this.removeEmailAddressesThatHaveBlacklistedDomain(data.replyTo, this.domainBlacklist);
```

Comment:
It looks like you're always passing `this.domainBlacklist` reference as a blacklist - if that's true then you don't event need this param as the `removeEmailAddressesThatHaveBlacklistedDomain()` can access it by itself as it's a method of the same class. It would make sense if you would make a separate function for that purpose, but in this case it seems redundant. But it's not a mistake so this is not a change request, I just feel the code would be a bit easier to read this way.

Reply:
right I thought I changed it but it seems I forgot